File: leetcode/1528.cpp
// Runtime: 11 ms, faster than 69.42% of C++ online submissions for Shuffle String.
// Memory Usage: 15.3 MB, less than 43.22% of C++ online submissions for Shuffle String.
class Solution {
 public:
  string restoreString(string s, vector<int> &indices) {
    string res(s);
    for (int i = 0; i < s.size(); ++i) {
      res[indices[i]] = s[i];
    }
    return res;
  }
};

File: leetcode/1528.py
"""
Runtime: 48 ms, faster than 94.56% of Python3 online submissions for Shuffle String.
Memory Usage: 14.4 MB, less than 21.72% of Python3 online submissions for Shuffle String.
"""


class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        sz = len(indices)
        d = {indices[i]: i for i in range(sz)}
        return "".join([s[d[i]] for i in range(sz)])
